test/extensions/network/dns_resolver/apple/apple_dns_impl_test.cc

```diff
@@ -292,6 +319,125 @@ TEST_F(AppleDnsImplTest, DnsIpAddressVersionV6Only) {
   dispatcher_->run(Event::Dispatcher::RunType::Block);
 }
 
+// For the tests below:
+//  - ipv4.google.com DNS records only resolve to IPv4 addresses, but if on a network that supports
+//    NAT64/DNS64, DNS can resolve to virtualized IPv6 addresses.
+//  - ipv6.google.com DNS records only resolve to IPv6 addresses.
```

Comment:
@RyanTheOptimist I agree, it's not ideal to rely on the actual network for these tests.  I debated whether I should add these particular tests or not; the reason I went with it is that (1) these tests already make network requests to do `google.com` DNS resolution, so I'm just adding to that with these tests and (2) I can mock what I _think_ should be the behavior from the Apple `DnSServiceGetAddrInfo` call but it's not guaranteed to be the way the Apple call actually behaves, which is why I think these tests are quite valuable.